atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -788,12 +788,17 @@ private static LockAndTimestampServices createRawServicesFromTimeLock(
             String userAgent) {
         Supplier<ServerListConfig> serverListConfigSupplier =
                 getServerListConfigSupplierForTimeLock(config, runtimeConfigSupplier);
-        TimeLockMigrator migrator =
-                TimeLockMigrator.create(metricsManager,
-                        serverListConfigSupplier, invalidator, userAgent, config.initializeAsync());
+
+        LockAndTimestampServices lockAndTimestampServices =
+                getLockAndTimestampServices(metricsManager, serverListConfigSupplier, userAgent);
+
+        TimeLockMigrator migrator = TimeLockMigrator.create(
+                lockAndTimestampServices.timestampManagement(),
+                invalidator,
+                config.initializeAsync());
         migrator.migrate(); // This can proceed async if config.initializeAsync() was set
-        return ImmutableLockAndTimestampServices.copyOf(
-                getLockAndTimestampServices(metricsManager, serverListConfigSupplier, userAgent))
+
+        return ImmutableLockAndTimestampServices.copyOf(lockAndTimestampServices)
```

Comment:
This changes the order the objects are created in, so the lock and timestamp services come before the migrator. `getLockAndTimestampServices` involves a bunch of proxies and/or thin wrappers above them (e.g. timelock-timestamp service adapter, leasing timelock client). It doesn't look like these are/can be accessed statically, so this should be fine.